src/boundaries/edgetracking.jl

```diff
@@ -0,0 +1,304 @@
+never actually converge to a point but (after a burn-in) continue populating the set
+consituting the edge state by tracking along it.
+
+A central idea behind this algorithm is that basin boundaries are typically the stable
+manifolds of unstable sets, namely edge states or saddles. The flow along the basin boundary 
+will thus lead to these sets, and the iterative bisection neutralizes the unstable
+direction of the flow away from the basin boundary. If the system possesses multiple edge 
+states, the algorithm will find one of them depending on where the initial bisection locates
+the boundary.
+
+## Output
+
+Returns a data type `EdgeTrackingResults` containing five fields:
+* `edge::StateSpaceSet`: the pseudo-trajectory representing the tracked edge segment
+  (given by the average in state space between `track1` and `track2`)
+* `track1::StateSpaceSet`: the pseudo-trajectory tracking the edge within basin 1
+* `track2::StateSpaceSet`: the pseudo-trajectory tracking the edge within basin 2
+* `time::Vector`: time points of the above `StateSpaceSet`s
+* `bisect_idx::Vector`: indices of `time` at which a re-bisection occurred
+* `success::Bool`: indicates whether the edge tracking has been successful or not
+
+Sometimes, the AttractorMapper used in the algorithm may erroneously identify both states
+`u1` and `u2` with the same basin of attraction due to being very close to the basin
+boundary. If this happens, a warning is raised and `EdgeTrackingResults.success = false`.
+"""
+function edgetracking(ds::DynamicalSystem, attractors::Dict;
+    bisect_thresh=1e-6,
+    diverge_thresh=1e-5,
+    u1=collect(values(attractors))[1][1],
+    u2=collect(values(attractors))[2][1],
+    maxiter=100,
+    abstol=0.0,
+    T_transient=0.0,
+    tmax=Inf,
+    Δt=0.01,
+    ϵ_mapper=nothing,
+    show_progress=true,
+    kwargs...)
+    
+    pds = ParallelDynamicalSystem(ds, [u1, u2])
+    mapper = AttractorsViaProximity(ds, attractors, ϵ_mapper; kwargs...)
+    
+    edgetracking(pds, mapper;
+        bisect_thresh, diverge_thresh, maxiter, abstol, T_transient, Δt, tmax, show_progress)
+end;
+
+"""
+    edgetracking(pds::ParallelDynamicalSystem, mapper::AttractorMapper; kwargs...)
+Low-level function for running the edge tracking algorithm, see [`edgetracking`](@ref)
+for a description, keyword arguments and output type.
+
+`pds` is a `ParallelDynamicalSystem` with two states. The `mapper` must be an
+`AttractorMapper` of subtype `AttractorsViaProximity` or `AttractorsViaRecurrences`.
+"""
+function edgetracking(pds::ParallelDynamicalSystem, mapper::AttractorMapper;
+    bisect_thresh=1e-6,
+    diverge_thresh=1e-5,
+    maxiter=100,
+    abstol=0.0,
+    T_transient=0.0,
+    Δt=0.01,
+    tmax=Inf,
+    show_progress=true)
+    
+    if bisect_thresh >= diverge_thresh
+        error("diverge_thresh must be larger than bisect_thresh.")
+    end
+
+    # initial bisection
+    u1, u2, success = bisect_to_edge(pds, mapper; bisect_thresh)
+    if ~success
```

Comment:
```suggestion
    if !success
```
`!` is more typical for boolean not.